fpna/DataModel/DataStructure.ipynb

```diff
@@ -0,0 +1,253 @@
+            "source": [
+                "public record BoundedCashflowDefinition : SimpleCashflowDefinition",
+                "\n{",
+                "\n    public double Minimum { get; init; }",
+                "\n",
+                "\n    public double Maximum { get; init; }",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "#### Referenced"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public record ReferencedCashflowDefinition : SimpleCashflowDefinition",
+                "\n{",
+                "\n    [IdentityProperty]",
+                "\n    [Dimension(typeof(AmountType))] ",
+                "\n    public string ReferenceAmountType { get; init; }",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public record FpaVariable : RawVariable",
+                "\n{",
+                "\n    //AocType => the minimum set for ifrs17 would be CL,C. (Projection of NB is included in CL,C)",
+                "\n    [Dimension(typeof(string))] ",
+                "\n    public string Layer { get; init; }",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public record FpaReportVariable",
+                "\n{",
+                "\n    [NotVisible] /*[Dimension(typeof(GroupOfContract))]*/",
+                "\n    public string DataNode { get; init; }",
+                "\n    ",
+                "\n    [NotVisible] [AggregateBy] /* [Dimension(typeof(AocType))]*/",
+                "\n    public string AocType { get; init; }",
+                "\n    ",
+                "\n    [NotVisible] /*[Dimension(typeof(Novelty))]*/",
+                "\n    public string Novelty { get; init; }",
+                "\n        ",
+                "\n    [NotVisible] [Dimension(typeof(AmountType))]",
+                "\n    public string AmountType { get; init; }",
+                "\n    ",
+                "\n    [NotVisible] /*[Dimension(typeof(EstimateType))]*/",
+                "\n    public string EstimateType { get; init; }",
+                "\n    ",
+                "\n    [NotVisible] [Dimension(typeof(string), nameof(Layer))] ",
+                "\n    public string Layer { get; init; }",
+                "\n",
+                "\n    [NotVisible] [Dimension(typeof(int), nameof(Index))]",
+                "\n    public int Index { get; init; }",
```

Comment:
Following our general approach of attaching business meaning to properties, Index doesn't have a business meaning. Year, Projection, etc... are better names at this stage, as this is supposed to be close to the business (report variable is the end of the process)

Reply:
renamed to Projection